### PR TITLE
Add `-rename_tables` option for DropSources

### PR DESCRIPTION
**Pull request by tomkrouper**
The purpose of this is to allow `DropSources` to run a `rename table` instead of a `drop table`. This is helpful for larger tables that dropping could lock MySQL and affect replication.

- [x] Implement initial rename table logic
- [x] test locally, verify rename table works
- [ ] add testing for rename table option
- [ ] add table name length check for rename
- [ ] add table existence check for rename
- [ ] ? add prefix/suffix option for rename table (maybe TODO later)

### CODE REVIEW

**Comment by tomkrouper** [comment]
Errors in `unit / build (mysql57)` start with:
```
E0623 17:24:21.502352   27620 shard_sync.go:87] Shard watch failed: node doesn't exist: keyspaces/ks/shards/-40/Shard
E0623 17:24:21.502801   27620 shard_sync.go:107] Failed to sync shard record: node doesn't exist: keyspaces/ks/shards/-40/Shard
```

Not sure how my work would have been involved in this at all.

File: go/vt/mysqlctl/s3backupstorage/s3_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package s3backupstorage
```
Comment (systay) [inline go/vt/mysqlctl/s3backupstorage/s3_test.go:1]: License header missing

File: go/vt/vtgate/executor_test.go [hunk 2]
```diff
@@ -1869,6 +1869,54 @@ func TestGenerateCharsetRows(t *testing.T) {
...
+
+	testMaxPayloadSizeOverride := []string{
+		"select /*vt+ IGNORE_MAX_PAYLOAD_SIZE=1 */ * from main1",
+		"insert /*vt+ IGNORE_MAX_PAYLOAD_SIZE=1 */ into main1(id) values (1), (2)",
+		"update /*vt+ IGNORE_MAX_PAYLOAD_SIZE=1 */ main1 set id=1",
+		"delete /*vt+ IGNORE_MAX_PAYLOAD_SIZE=1 */ from main1 where id=1",
+	}
+	for _, query := range testMaxPayloadSizeOverride {
+		_, err := executor.Execute(context.Background(), "TestExecutorMaxPayloadSizeWithOverride", session, query, nil)
+		assert.Equal(t, nil, err, "err should be nil")
```
Comment (systay) [inline go/vt/vtgate/executor_test.go:1908]: nit: I quite like using `assert.NoError` for these cases. This works, so it's only about my OCD :)

File: go/vt/vttablet/tabletserver/tabletenv/config_test.go [hunk 3]
```diff
@@ -147,3 +151,118 @@ func TestClone(t *testing.T) {
 	cfg1.OltpReadPool.Size = 10
 	assert.NotEqual(t, cfg1, cfg2)
 }
+
+func TestFlags(t *testing.T) {
```
Comment (systay) [inline go/vt/vttablet/tabletserver/tabletenv/config_test.go:155]: Wouldn't this test benefit from being a "table-driven test"?

**Comment by tomkrouper** [comment]
@systay, thanks for the review comments. I must have messed up merging somehow, because none of those changes are things I did.

File: go/vt/vtctl/vtctl.go [hunk 4]
```diff
@@ -315,7 +315,7 @@ var commands = []commandGroup{
 				"[-cell=<cell>] [-tablet_types=<source_tablet_types>] -workflow=<workflow> <source_keyspace> <target_keyspace> <table_specs>",
 				`Move table(s) to another keyspace, table_specs is a list of tables or the tables section of the vschema for the target keyspace. Example: '{"t1":{"column_vindexes": [{""column": "id1", "name": "hash"}]}, "t2":{"column_vindexes": [{""column": "id2", "name": "hash"}]}}`},
 			{"DropSources", commandDropSources,
-				"[-dry_run] <keyspace.workflow>",
+				"[-dry_run] [-rename_table] <keyspace.workflow>",
```
Comment (sougou) [inline go/vt/vtctl/vtctl.go:318]: `rename_tables`

File: go/vt/vtctl/vtctl.go [hunk 5]
```diff
@@ -2042,6 +2042,7 @@ func commandMigrateServedFrom(ctx context.Context, wr *wrangler.Wrangler, subFla
 
 func commandDropSources(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.FlagSet, args []string) error {
 	dryRun := subFlags.Bool("dry_run", false, "Does a dry run of commandDropSources and only reports the actions to be taken")
+	renameTable := subFlags.Bool("rename_table", false, "Rename tables instead of dropping them")
```
Comment (sougou) [inline go/vt/vtctl/vtctl.go:2045]: rename_tables

File: go/vt/vtctl/vtctl.go (hunk 4 above)
Comment (tomkrouper) [reply go/vt/vtctl/vtctl.go]: fair nit, fixed.

File: go/vt/vtctl/vtctl.go (hunk 5 above)
Comment (tomkrouper) [reply go/vt/vtctl/vtctl.go]: fixed

**Comment by tomkrouper** [comment]
deleted because seeing all the other changes doesn't seem right to me. I'm used to seeing only my changes in the PR. I assume my workflow works great internally while working on a local repo, but is wrong for forks.

**Comment by shlomi-noach** [comment]
@tomkrouper if you like, I'm happy to work with you to figure out how to make this PR work.

**Pull request closed without merging**